### PR TITLE
CRNS-156/ChannelPreview

### PR DESCRIPTION
**Pull request by DanC5**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/ChannelList/utils.ts [hunk 1]
```diff
@@ -6,26 +6,26 @@ import type {
...
 
 type MoveParameters<
-  At extends UnknownType = UnknownType,
-  Ch extends UnknownType = UnknownType,
-  Co extends string = LiteralStringForUnion,
-  Ev extends UnknownType = UnknownType,
-  Me extends UnknownType = UnknownType,
-  Re extends UnknownType = UnknownType,
-  Us extends UnknownType = UnknownType
+  At extends UnknownType = DefaultAttachmentType,
```
Comment (virdesai) [inline src/components/ChannelList/utils.ts:9]: were these defaults already being imported here?

Comment (DanC5) [reply src/components/ChannelList/utils.ts:9]: oh missed that one on the imports, did a global replace all

File: src/components/ChannelList/ChannelListMessenger.tsx [hunk 2]
```diff
@@ -230,52 +149,46 @@ const ChannelListMessenger = <
...
+  const renderItem = (channel: Channel<At, Ch, Co, Ev, Me, Re, Us>) => (
+    <ChannelPreview<At, Ch, Co, Ev, Me, Re, Us> {...props} channel={channel} />
+  );
+
+  const renderChannels = () => (
+    <>
+      {renderHeaderIndicator()}
+      <FlatList
+        data={channels}
+        extraData={forceUpdate}
```
Comment (virdesai) [inline src/components/ChannelList/ChannelListMessenger.tsx:161]: I know it's been in there but forceUpdate seems like an odd name for extraData

File: src/components/ChannelList/ChannelListMessenger.tsx [hunk 3]
```diff
@@ -230,52 +149,46 @@ const ChannelListMessenger = <
...
+        keyExtractor={(item) => item.cid}
+        ListEmptyComponent={() => <EmptyStateIndicator listType='channel' />}
+        ListFooterComponent={() =>
+          loadingNextPage ? <FooterLoadingIndicator /> : null
+        }
+        onEndReached={loadNextPage}
+        onEndReachedThreshold={loadMoreThreshold}
+        onRefresh={refreshList}
+        ref={(flRef) => {
```
Comment (nhannah) [inline src/components/ChannelList/ChannelListMessenger.tsx]: want to name this better?

File: src/components/ChannelPreview/hooks/useChannelPreviewDisplayAvatar.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    };
+  } else if (currentUserId) {
+    const members = Object.values(channel.state?.members || {});
+    const otherMembers = members.filter(
+      (member) => member.user?.id !== currentUserId,
+    );
+
+    if (otherMembers.length === 1) {
+      return {
+        image: otherMembers[0].user?.image.asMutable() as string,
```
[the remark this answers is not in the document]
Comment (DanC5) [reply src/components/ChannelPreview/hooks/useChannelPreviewDisplayAvatar.ts]: changed the gennies around to fix this

File: src/components/ChannelPreview/ChannelPreviewMessenger.tsx [hunk 5]
```diff
@@ -83,47 +136,28 @@ const ChannelPreviewMessenger = ({
...
-  unread: PropTypes.number,
-};
-
 ChannelPreviewMessenger.themePath = 'channelPreview';
 
 // TODO: remove HOC and use a theme context provider
-export default themed(ChannelPreviewMessenger);
+export default themed(
+  ChannelPreviewMessenger,
+) as typeof ChannelPreviewMessenger;
```
[the remark this answers is not in the document]
Comment (DanC5) [reply src/components/ChannelPreview/ChannelPreviewMessenger.tsx:163]: the generics get lost when passed into the function, same as with React.memo on ChannelPreview so had to cast the type explicitly

File: src/components/ChannelPreview/hooks/useLatestMessagePreview.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  channel: Channel<At, Ch, Co, Ev, Me, Re, Us>,
+  lastMessage:
```
Comment (nhannah) [inline src/components/ChannelPreview/hooks/useLatestMessagePreview.ts:158]: Can you just do lastMessage?:

**Review by nhannah — APPROVED** [review]

**Comment by nhannah** [comment]
Great 🥕 👨‍🌾

**Pull request merged**